### ISSUE TITLE
Using cv2.decolor on black-and-white image outputs all zero image

### ISSUE THREAD

**Issue by John-Ao** [opencv/opencv#22002]
##### System information (version)
- Python => 3.9.6
- opencv-python => 4.5.5
- numpy => 1.20.0
- Operating System / Platform => Windows 11

##### Detailed description

Using `cv2.decolor` on black and white image outputs all zero image. It can be reproduced with the following code.

Code:

```python
import cv2
import numpy as np

print(cv2.__version__)
print(np.__version__)
# create a black-and-white image
im = np.ones((300, 300, 3), np.uint8) * 255
im[100:200, 100:200] = 0
# call decolor
im = cv2.decolor(im)[1]
print(np.all(im == 0))
```

Output:
```
4.5.5
1.20.0
True
```

**Comment by sturkmen72** [comment]
confirmed!
also too bright input images 
input
![original](https://github.com/user-attachments/assets/091d2b0e-02ca-4f2d-bbc9-caf523b9587d)
output grayscale
![grayscale](https://github.com/user-attachments/assets/09ee2557-8a03-4d4d-915b-e261015defa9)
output color_boost
![color_boost](https://github.com/user-attachments/assets/9b7ddb79-4491-4f29-8a9f-c19d1fd705ec)

**Comment by sturkmen72** [comment]
when the input image is slightly less brighter
input
![original](https://github.com/user-attachments/assets/d2cc88dc-6ad6-4465-9ef8-2ad5cba87b9d)
output grayscale
![grayscale](https://github.com/user-attachments/assets/47b5113e-c180-4269-8aa7-902a810a2911)
output color_boost
![color_boost](https://github.com/user-attachments/assets/f8176ec9-9c68-4e00-ba19-5fdb7d35abae)

**Comment by sturkmen72** [comment]
this issue can be labeled as "good first issue"

**Comment by huneenzahid** [comment]
Hello, I’m a first-year CS student and new to OpenCV.
I saw that this issue was recently confirmed and suggested as a good first issue.
I’d like to try reproducing it on the latest OpenCV version and investigate further.
Any guidance on where to start would be appreciated.

**Comment by huneenzahid** [comment]
I tested this on OpenCV 4.13.0 with Python 3.13, and the issue still reproduces.

Using a simple black-and-white image, cv2.decolor() returns an all-zero grayscale image.

I can confirm the bug persists on the latest version.

**Comment by deepak-shenoy** [comment]
I don't think that this is an issue.  Note that cv2.decolor(im)[1] calls OpenCV's decolor function, which converts a color image to grayscale while trying to preserve contrast. It returns a tuple of two images:

[0] — the grayscale result
[1] — a "boosted" color image (a re-colored version)

So [1] grabs the second output (the color boost image), discarding the grayscale result.

Because cv2.decolor expects a color image with actual color variation. When you feed it a purely black-and-white image (white background with a black square — no hue differences, just intensity), the algorithm has nothing meaningful to work with for its contrast-preserving grayscale conversion.

The result is that [1] (the color boost output) comes back as all zeros (black), which is why np.all(im == 0) prints True.

The issue isn't about display — it's that decolor produces a blank (all-black) image when given input with no actual color information. To see a proper grayscale version of your black-and-white square, you'd just use:

im = cv2.cvtColor(im, cv2.COLOR_BGR2GRAY)
